### DISCUSSION TITLE
Firmware version

### DISCUSSION THREAD

**Discussion by demian054** [TillFleisch/ESPHome-HLK-LD2450#30] [Q&A]
is there any firmware requirement? 

I'm trying to use this custom component, but I'm getting those messages
```
Clearing RX buffer.
Sensor is not sending updates, attempting to leave config mode.
Sending command timed out! Is the sensor connected?
Clearing RX buffer.
```

note: using the Android app I can read the values

**Comment by TillFleisch** [comment]
Yes, there is a firmware requirement for some functionality (like buttons and switches), as these option were added to the sensor at a later stage.
I am currently using `V2.04.23101915`. Unfortunately, I can't recall what the minimum requirement is (probably anything starting with V2..). You can update the sensors firmware using the official HLK LD2450 App.

Make sure to go to though the [troubleshooting section](https://github.com/TillFleisch/ESPHome-HLK-LD2450?tab=readme-ov-file#troubleshooting), in particular, try to use 5V and reliable connections. Long wires and improper wiring are known to cause this kind of issue. Try running a `minimal configuration` which does not include any other sensors/components.
